### PR TITLE
update e2e coverage

### PR DESCRIPTION
**Pull request by ashetty1**
Covers #430

### CODE REVIEW

File: tests/e2e/README.md [hunk 1]
```diff
@@ -17,20 +17,31 @@ Run `make test-e2e` to execute the tests.
...
+| `component get` | Y |
+| `component set` | Y |
 | `create --git` | N |
 | `create --local` | Y |
+| `create --binary` | N |
+| `delete` | N |
+| `describe` | N |
+| `link` | Y |
 | `list` | Y |
 | `push` | Y |
```
Comment (kadel) [inline tests/e2e/README.md]: are we covering push for every component type (--git/--local/--binary)?

Comment (ashetty1) [reply tests/e2e/README.md]: Will split and update this bit.

File: tests/e2e/README.md [hunk 2]
```diff
@@ -17,20 +17,33 @@ Run `make test-e2e` to execute the tests.
...
+| `app describe` | N |
 | `app list` | Y |
-| `catalog` | N |
-| `create --git` | N |
+| `catalog list` | Y |
+| `catalog search` | N |
+| `component get` | Y |
+| `component set` | Y |
+| `create --git` | Y |
 | `create --local` | Y |
```
Comment (kadel) [inline tests/e2e/README.md]: there is no test calling `odo create --local`

File: tests/e2e/README.md [hunk 3]
```diff
@@ -17,20 +17,33 @@ Run `make test-e2e` to execute the tests.
...
 | `app delete` | Y |
+| `app describe` | N |
 | `app list` | Y |
-| `catalog` | N |
-| `create --git` | N |
+| `catalog list` | Y |
+| `catalog search` | N |
+| `component get` | Y |
+| `component set` | Y |
+| `create --git` | Y |
```
Comment (kadel) [inline tests/e2e/README.md]: this is currently in progress in https://github.com/redhat-developer/odo/pull/437

File: tests/e2e/README.md [hunk 4]
```diff
@@ -17,20 +17,33 @@ Run `make test-e2e` to execute the tests.
...
-| `storage list` | N |
-| `storage remove` | N |
-| `update` | N |
+| `push` --local | Y |
+| `push` --git | Y |
+| `push` --binary | Y |
+| `storage create` | Y |
+| `storage list` | Y |
+| `storage delete` | N |
+| `update` | Y |
```
Comment (kadel) [inline tests/e2e/README.md]: we need to track tests for all directions.
We have try types of component source and we need to tests all posible update permutations.


from `--git` to `--local`
from `--local` to `--git`
from `--binary` to `--git`
from `--git` to `--binary`
from `--local` to `--binary`
from `--binary` to `--local`

File: tests/e2e/README.md (hunk 2 above)
Comment (ashetty1) [reply tests/e2e/README.md]: Oh I think #437 replaces https://github.com/redhat-developer/odo/blob/master/tests/e2e/e2e_test.go#L192

I think I will undo that change and add a separate test for `create --git`?

File: tests/e2e/README.md (hunk 4 above)
Comment (ashetty1) [reply tests/e2e/README.md]: Sure, will add them

File: tests/e2e/README.md (hunk 2 above)
Comment (kadel) [reply tests/e2e/README.md]: We should cover all uses.
We have 4 different commands that can create a component.

`odo create`
`odo create --local`
`odo create --binary`
`odo create --git`

The first two are doing almost the same, but we should have tests covering both of them.

**Comment by cdrage** [comment]
There's an awesome VIM plugin for markdown here: https://github.com/plasticboy/vim-markdown where you can use `:TableFormat` to format the table with appropriate spacing!

Other than @kadel 's comments (add the update --git, etc. testing), LGTM.

**Comment by kadel** [comment]
My main issue with this PR is that it marks stuff as covered that is not yet merged or can't be merged yet (#437)

**Comment by anushshetty** [comment]
Will look into #437 today.

**Comment by ashetty1** [comment]
Will undo that part and resend

**Comment by cdrage** [comment]
Any updates to this @ashetty1 ? :+1:

**Comment by ashetty1** [comment]
@cdrage the PR has been updated will all the changes requested.

**Comment by cdrage** [comment]
This LGTM :+1: Since it's a doc-change, only 1 ACK needed. I'm going to merge!

**Pull request merged**